servicemigration/context.py

```diff
@@ -52,12 +70,28 @@ def commit(self, filename=None):
             else:
                 serviceList.append(serial)
         data = {
-            "modified": serviceList,
-            "cloned": cloneList
+            "ServiceID": self.getTopService()._Service__data["ID"],
+            "Modified": serviceList,
+            "Cloned": cloneList
         }
-        f = open(filename, 'w')
-        f.write(json.dumps(data, indent=4, sort_keys=True))
-        f.close()
+        data = json.dumps(data, indent=4, sort_keys=True)
+        cpClient = None
+        if ZenUtils:
```

Comment:
Lines 79-80 are premature in the sense that you don't need cpClient until line 92, so there's no need to initialize it before it's used.

Lines 82-84 and 88-90 are basically duplicates.

So something like this would be a little more concise, but hopefully still conveys the 3 possible outcomes fairly clearly:
```
if filename is not None:
   outfile = filename
else:
   outfile = os.environ["MIGRATE_OUTPUTFILE"] if "MIGRATE_OUTPUTFILE" in os.environ else None

if outfile is not None:
    f = open(outfile, 'w')
    f.write(data)
    f.close()
elif ZenUtils:
   cpClient = ControlPlaneClient(**getConnectionSettings())
   cpClient.postServicesForMigration(data)
else:
    raise ValueError("Can't find migration output location.")
```